pkg/library/reconcilers/target_ref_reconciler.go

```diff
@@ -209,3 +211,35 @@ func (r *TargetRefReconciler) ReconcileGatewayPolicyReferences(ctx context.Conte
 
 	return nil
 }
+
+func (r *TargetRefReconciler) UpdateGatewayCondition(ctx context.Context, condition metav1.Condition, gatewayDiff *GatewayDiffs) error {
+	// update condition if needed
+	gatewayDiffs := append(gatewayDiff.GatewaysWithValidPolicyRef, gatewayDiff.GatewaysMissingPolicyRef...)
+	for i, gw := range gatewayDiffs {
+		previous := gw.DeepCopy()
+		gwCondition := condition.DeepCopy()
+		gwCondition.ObservedGeneration = gw.Generation
+		meta.SetStatusCondition(&gatewayDiffs[i].Status.Conditions, *gwCondition)
+		if !reflect.DeepEqual(previous.Status.Conditions, gw.Status.Conditions) {
+			if err := r.Client.Status().Update(ctx, gw.Gateway); err != nil {
+				return err
+			}
+			// TODO(guicassolato): for inherited policies only – add the policyaffetced status condition on all routes accepted by the gateway
```

Comment:
This is assuming that, no matter at level the policy targets an object (gateway _or_ route), all objects (gateways _and_ routes) will be marked as affected by the policy.

Is that really what we want or we want to set this condition only on the object targeted by the policy plus its children (but never upwards)?

cc @maleck13 @R-Lawton @david-martin

Reply:
I would go for the latter. It makes more sense to me this way

Reply:
Ok. I need to change a few things then. Thanks!

Reply:
Changed so direct policies (TLS/DNS) only add to the statuses of the objects they target (Gateways only) and inherited (auth/RL) to the objects they target plus children beneath.